packages/zoltan2/src/input/Zoltan2_Adapter.hpp

```diff
@@ -159,9 +169,16 @@ template <typename User>
   // *   getNumWeightsPerID > 0.
   // *   This function should not be called if getNumWeightsPerID is zero.
   // */ 
-  virtual void getWeightsKokkosView(Kokkos::View<scalar_t *> &/* wgt */, 
-                              int /* idx */ = 0) const {
-    Z2_THROW_NOT_IMPLEMENTED
+  virtual void getWeightsKokkosView(Kokkos::View<scalar_t *,
+    typename node_t::device_type> &wgt, int idx) const {
+    throw std::logic_error("getWeightsKokkosView not implemented.");
+  }
+
+  // TODO: 2nd form returns the 2darray - still need to decide how this
```

Comment:
I removed this TODO. Currently we have getWeightsKokkosView and a getWeightsKokkosView2D. Only two things depend on getWeightsKokkosView, the tests Zoltan2_BasicKokkosIdentifierInput and Zoltan2_kokkosBlock. Zoltan2_BasicKokkosIdentifierInput would easily be refactored to use the 2D form, which MJ also depends on. Zoltan2_kokkosBlock works through Zoltan2::IdentifierModel which calls getWeightsView, so depends on it then calling getWeightsKokkosView. Maybe the role of these tests could be rethought. I am wondering if we would like to remove the getWeightsKokkosView completely. Then the getWeightsView method here would throw instead of calling getWeightsKokkosView. It depends on whether we want to preserve Zoltan2_kokkosBlock in the present form.